src/app/pages/BibPage.jsx

```diff
@@ -43,61 +43,51 @@ export const BibPage = (
+  // then stop fetching more items.
   useEffect(() => {
-    if (bib && bib.fetchMoreItems) {
-      checkForMoreItems();
+    if (bib && bib.fetchMoreItems || showAll) {
+      checkForMoreItems(showAll);
     }
-  }, [bib, checkForMoreItems]);
+  }, [bib, checkForMoreItems, showAll]);
 
   /*
    * Checks to see if we need to fetch more items. If the "View All Items"
    * button is clicked, we want to make multiple batch requests to get all
-   * the items. If the pagination component is used, we only want to make
-   * one request for the next X items through the `once` argument.
+   * the items.
    */
-  const checkForMoreItems = useCallback((once = false) => {
+  const checkForMoreItems = useCallback((showAll = false) => {
     if (!bib || !bib.items || !bib.items.length || (bib && bib.done)) {
       // nothing to do
-    } else if (bib && bib.items.length >= numItemsMatched) {
-      // Once we have fetched all the items, we're done,
-      // so stop fetching more items.
-      // `fetchMoreItems` is used to trigger the useEffect but
-      // `done` is used to stop the API requests.
+    } else if (bib && bib.items.length >= numItemsMatched || !showAll) {
+      // 1. Once we have fetched all the items, we're done, so stop fetching
+      //   more items. `fetchMoreItems` is used to trigger the useEffect but
+      //   `done` is used to stop the API requests.
+      // 2. If `showAll` is false, stop fetching more items. This can happen
+      //   if the patron clicks on another bib page or performs a new filter
+      //   search.
       dispatch(
         updateBibPage({
           bib: Object.assign({}, bib, { done: true, fetchMoreItems: false })
         })
       );
     } else {
       // We need to fetch for more items.
-      const searchStr = window.location.search;
+      const searchStr =
+        window.location.search ? `&${window.location.search.substring(1)}` : '';
       const baseUrl = appConfig.baseUrl;
-      const itemFrom = bib.itemFrom || itemBatchSize;
-      let filterQuery = '';
-
-      // If there are filters, we need to add them to the API query.
-      if (searchStr) {
-        const searchParams = new URLSearchParams(searchStr);
-        for (const [qKey, qValue] of searchParams) {
-          // For the date filter, we use the direct value.
-          if (qKey === 'date') {
-            filterQuery += `&${qKey}=${qValue}`;
-          } else if (qKey !== "itemPage") {
-            // For other filters, we need to map the label to the id.
-            filterQuery += `&${qKey}=${fieldToOptionsMap[qKey][qValue]}`;
-          }
-        }
-      }
+      const itemFrom = !bib.itemFrom ? 0 : bib.itemFrom;
 
       // Fetch the next batch of items using the `itemFrom` param.
       const bibApi = `${window.location.pathname.replace(
         baseUrl,
         `${baseUrl}/api`,
-      )}?itemFrom=${itemFrom}${filterQuery}`;
+      )}?items_from=${itemFrom}${searchStr}`;
 
       ajaxCall(
```

Comment:
If it's not a huge lift, can we possibly remove this ajax call? i see that the callback updatebibpage might complicate doing that, but i think it would be a good move.

Reply:
Remove it from this file or entirely? It's not entirely apparent to me how we'll fetch more items without the ajax call. Do you mean using the router push update to handle this instead?

Reply:
Moved this function to a separate util file and made it a custom react hook. No tests yet because enzyme...